Helm 3.19 crashes with a nil pointer dereference

### What happened?

```
❯ helm template otlp_forwarder ./helmchart 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x55fbaa5c74d6]

goroutine 1 [running]:
google3/third_party/golang/helm_sh/helm/v/v3/pkg/action/action.(*Install).RunWithContext(0x1037f789e000, {0x55fbaae459c0?, 0x2?}, 0x1037f7aebae0, 0x1037f7add5c0)
        third_party/golang/helm_sh/helm/v/v3/pkg/action/install.go:173 +0x1f6
main.runInstall({0x1037f7633d00, 0x2, 0x2}, 0x1037f789e000, 0x1037f78d45a0, {0x55fbaaf59f08, 0x1037f75bc028})
        third_party/golang/helm_sh/helm/v/v3/cmd/helm/install.go:141 +0x4f2
main.newTemplateCmd.func2(0x1037f7522500?, {0x1037f7633d00, 0x2, 0x2})
        third_party/golang/helm_sh/helm/v/v3/cmd/helm/template.go:95 +0x29f
google3/third_party/golang/cobra/cobra.(*Command).execute(0x1037f7ae6008, {0x1037f7633cc0, 0x2, 0x2})
        third_party/golang/cobra/command.go:985 +0xb42
google3/third_party/golang/cobra/cobra.(*Command).ExecuteC(0x1037f79d8f08)
        third_party/golang/cobra/command.go:1117 +0x44f
google3/third_party/golang/cobra/cobra.(*Command).Execute(...)
        third_party/golang/cobra/command.go:1041
main.main()
        third_party/golang/helm_sh/helm/v/v3/cmd/helm/helm.go:62 +0xc5
```



### What did you expect to happen?

I expected that Helm prints an error message instead of the crash.

### How can we reproduce it (as minimally and precisely as possible)?

When I run `helm template ./helmchart` instead of `helm template otlp_forwarder ./helmchart` I get this:

```
❯ helm template  ./helmchart              
Error: template: otlp-forwarder/templates/serviceaccount.yaml:4:11: executing "otlp-forwarder/templates/serviceaccount.yaml" at <include "otlp_forwarder.serviceAccountName" .>: error calling include: template: otlp-forwarder/templates/_helpers.tpl:68:57: executing "otlp_forwarder.serviceAccountName" at <.Values.serviceAccount.name>: nil pointer evaluating interface {}.name

Use --debug flag to render out invalid YAML
```

Might be the root cause of the issue?

### Helm version

<details>

```console
$ helm version
version.BuildInfo{Version:"v3.19", GitCommit:"", GitTreeState:"", GoVersion:"go1.26-20251102-RC00 cl/827152512 +388c41c412 X:fieldtrack,boringcrypto"}
```
</details>


### Kubernetes version

<details>

```console
$ kubectl version
# paste output here
```

</details>

--- a/pkg/action/install.go
+++ b/pkg/action/install.go
@@ -186,32 +186,34 @@ func (i *Install) installCRDs(crds []chart.CRD) error {
 			return err
 		}
 
-		// If we have already gathered the capabilities, we need to invalidate
-		// the cache so that the new CRDs are recognized. This should only be
-		// the case when an action configuration is reused for multiple actions,
-		// as otherwise it is later loaded by ourselves when getCapabilities
-		// is called later on in the installation process.
-		if i.cfg.Capabilities != nil {
-			discoveryClient, err := i.cfg.RESTClientGetter.ToDiscoveryClient()
+		if i.cfg.RESTClientGetter != nil {
+			// If we have already gathered the capabilities, we need to invalidate
+			// the cache so that the new CRDs are recognized. This should only be
+			// the case when an action configuration is reused for multiple actions,
+			// as otherwise it is later loaded by ourselves when getCapabilities
+			// is called later on in the installation process.
+			if i.cfg.Capabilities != nil {
+				discoveryClient, err := i.cfg.RESTClientGetter.ToDiscoveryClient()
+				if err != nil {
+					return err
+				}
+				if discoveryClient != nil {
+					i.cfg.Log("Clearing discovery cache")
+					discoveryClient.Invalidate()
+					_, _ = discoveryClient.ServerGroups()
+				}
+			}
+
+			// Invalidate the REST mapper, since it will not have the new CRDs
+			// present.
+			restMapper, err := i.cfg.RESTClientGetter.ToRESTMapper()
 			if err != nil {
 				return err
 			}
-
-			i.cfg.Log("Clearing discovery cache")
-			discoveryClient.Invalidate()
-
-			_, _ = discoveryClient.ServerGroups()
-		}
-
-		// Invalidate the REST mapper, since it will not have the new CRDs
-		// present.
-		restMapper, err := i.cfg.RESTClientGetter.ToRESTMapper()
-		if err != nil {
-			return err
-		}
-		if resettable, ok := restMapper.(meta.ResettableRESTMapper); ok {
-			i.cfg.Log("Clearing REST mapper cache")
-			resettable.Reset()
+			if resettable, ok := restMapper.(meta.ResettableRESTMapper); ok {
+				i.cfg.Log("Clearing REST mapper cache")
+				resettable.Reset()
+			}
 		}
 	}
 	return nil

--- a/pkg/action/install_test.go
+++ b/pkg/action/install_test.go
@@ -894,6 +894,28 @@ func TestNameAndChartGenerateName(t *testing.T) {
 	}
 }
 
+func TestInstallCRDsWithNilRESTClientGetter(t *testing.T) {
+	config := actionConfigFixture(t)
+	failingKubeClient := kubefake.FailingKubeClient{PrintingKubeClient: kubefake.PrintingKubeClient{Out: io.Discard}, BuildDummy: true}
+	config.KubeClient = &failingKubeClient
+	config.RESTClientGetter = nil
+	instAction := NewInstall(config)
+
+	crds := []chart.CRD{{
+		Name: "test-crd",
+		File: &chart.File{
+			Name: "crds/test-crd.yaml",
+			Data: []byte("kind: CustomResourceDefinition"),
+		},
+	}}
+
+	var err error
+	require.NotPanics(t, func() {
+		err = instAction.installCRDs(crds)
+	})
+	require.NoError(t, err)
+}
+
 func TestInstallWithLabels(t *testing.T) {
 	is := assert.New(t)
 	instAction := installAction(t)

